### PR TITLE
[Java] CLI-backed Artifact Repository should pass-through HTTP auth

### PR DESCRIPTION
**Pull request by aarondav**
Requires #402 to actually work; this just passes the flags through the the mlflow CLI process so we can access artifactories which requires auth.

I only did unit testing here; we should have an end-to-end test confirming the Java API can read and write artifacts from authenticated backends.

### CODE REVIEW

File: mlflow/java/client/pom.xml [hunk 1]
```diff
@@ -113,8 +116,8 @@
               <shadedPattern>${mlflow.shade.packageName}.databricks</shadedPattern>
             </relocation>
             <relocation>
-              <pattern>org.yaml</pattern>
-              <shadedPattern>${mlflow.shade.packageName}.yaml</shadedPattern>
+              <pattern>org.ini4j</pattern>
```
Comment (aarondav) [inline mlflow/java/client/pom.xml]: Oopsie.

File: mlflow/java/client/pom.xml [hunk 2]
```diff
@@ -92,6 +94,7 @@
               <include>commons-io:commons-io</include>
               <include>commons-logging:commons-logging</include>
               <include>org.apache.httpcomponents:httpcore</include>
+              <include>org.ini4j:ini4j</include>
```
Comment (aarondav) [inline mlflow/java/client/pom.xml]: Forgot to include in shaded jar.

File: mlflow/java/client/pom.xml [hunk 3]
```diff
@@ -45,15 +45,6 @@
       <groupId>commons-io</groupId>
       <artifactId>commons-io</artifactId>
     </dependency>
-    <dependency>
```
Comment (aarondav) [inline mlflow/java/client/pom.xml]: I removed these deps as they were a bit much to annotate 4 params. We can add back as needed, but they would need to be shaded for all customers.

File: mlflow/java/client/src/main/java/org/mlflow/tracking/creds/DatabricksConfigHostCredsProvider.java [hunk 4]
```diff
@@ -68,7 +68,7 @@ private void reloadConfig() {
     String username = section.get("username");
     String password = section.get("password");
     String token = section.get("token");
-    boolean insecure = section.get("insecure", "false").equals("true");
+    boolean insecure = section.get("insecure", "false").toLowerCase().equals("true");
```
Comment (aarondav) [inline mlflow/java/client/src/main/java/org/mlflow/tracking/creds/DatabricksConfigHostCredsProvider.java]: Fixes andrew's comment from prior PR

**Review by smurching — APPROVED** [review]
LGTM!

**Pull request merged**